public/js/app/leaderboards.js

```diff
@@ -51,185 +39,130 @@ function leaderboardUpdate(data) {
-function changeLeaderboard(newLeaderboard) {
-  leaderboardOneJSON(newLeaderboard)
-    .then(data => {
-      pageNumber = 0;
-      
-
+//This function checks the different filters the user has chosen
+function timeCheck(timeSelected) {
+  timeFilter = timeSelected;
+  d = new Date();
+  minusTimeFilter = d.setMonth(d.getMonth() - timeFilter);
+  currentDate = new Date(minusTimeFilter).toISOString();
 
-      leaderboardUpdate(data);
-      //Makes the next and previous button revert to the first page status
-      nextButton.classList.remove('exhaustedLeaderboard');
-      previousButton.classList.add('exhaustedLeaderboard');
-      lastButton.classList.remove('exhaustedLeaderboard');
-      firstButton.classList.add('exhaustedLeaderboard');
+  //Re-insert all players that were kicked, to have a clean slate
+  for (let i = 0; i < timedOutPlayers.length; i++) {
+    playerList.push(timedOutPlayers[i]);
+  }
+  playerList.sort((a, b) => a.value.rating - b.value.rating);
+  timedOutPlayers = [];
 
-    });
-}
+  //kick all the players that dont meet the time filter
+  for (let i = 0; i < playerList.length; i++) {
 
-//Gets called once so it generates a leaderboard
-changeLeaderboard('1v1');
+    if (currentDate > playerList[i].date) {
 
-//This changes the current leaderboard depending on which leaderboard was clicked
-let leaderboards = ['1v1', '2v2', '4v4'];
-for (let k = 0; k < 3; k++) {
-  document.getElementById(`leaderboard${leaderboards[k]}`).addEventListener('click', () => {
-    addNextPlayer = 0;
-    plusOrMinusPlayerList = 1;
-    changeLeaderboard(`${leaderboards[k]}`);
-    //Remove the active class from all four buttons
-    for (let l = 0; l < 3; l++) {
-      document.getElementById(`leaderboard${leaderboards[l]}`).classList.remove('activeLeaderboard');
+      timedOutPlayers.push(playerList[i]);
+      playerList.splice(i, 1);
     }
-    //add the active class to the button clicked
-    document.getElementById(`leaderboard${leaderboards[k]}`).classList.add('activeLeaderboard');
-  });
+
+  }
+  pageChange(0);
 }
 
-//This function triggers when the next, previous, first or last button are clicked
-function changePage() {
-  leaderboardOneJSON(currentLeaderboard)
+// This changes the current leaderboard(newLeaderboard), sets the page to 0 (pageNumber = 0) and resets the next and previous buttons.
+function changeLeaderboard(newLeaderboard) {
+  leaderboardOneJSON(newLeaderboard)
     .then(data => {
-      leaderboardUpdate(data);
+      playerList = data;
+      let dataLengthString = (playerList.length).toString();
+      if (dataLengthString.length !== 4) {
+        lastPage = dataLengthString.slice(0, 1);
+      } else {
+        lastPage = dataLengthString.slice(0, 2);
+      }
+      console.log(playerList.length);
+      pageChange(0);
     });
 }
+//Gets called once so it generates a leaderboard
+changeLeaderboard('1v1');
```

Comment:
Use constant, no magic declaration in code
const defaultLeaderboardName = '1v1' in the beginning of your script